### PR TITLE
.github: Migrate Markdown issue templates to YAML forms

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
Preview at https://github.com/AnthonyLatsis/swift/issues/new/choose.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2023 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See https://swift.org/LICENSE.txt for license information
+# See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+blank_issues_enabled: true
+contact_links:
```
Comment (AnthonyLatsis) [inline .github/ISSUE_TEMPLATE/config.yml:10]: Very interested to hear your thoughts on what order of contact links makes the most sense.

**Comment by AnthonyLatsis** [comment]
> Spellcheck: "Begginer", "consice", "prefferably".

How embarrassing. Thank you!

> * 🪲 Report an issue on the Feedback Assistant app 

Yeah, I guess redirection redundancy cannot hurt. I feel like most people won’t read through to this item and go straight for "Report a bug" though.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (xedin) [reply .github/ISSUE_TEMPLATE/config.yml:10]: Maybe "Ask" and "Report" links should be moved up and "Learn Swift" should be last?

**Review by ahoppen — APPROVED** [review]
Looks good to me. Thanks!

**Review by LucianoPAlmeida — APPROVED** [review]
This looks awesome!

**Review by shahmishal — APPROVED** [review]
Love it!

**Comment by AnthonyLatsis** [comment]
Could someone help me out with capitalization? We are pretty inconsistent about how we capitalize most of the following compound nouns in all sorts of content from internal docs to TSPL and Swift.org, and I couldn’t find any helpful prescriptions in the Apple Style Guide. I’d like to get this right if possible.

* Swift forums or Swift Forums?
* Swift compiler or Swift Compiler?
* Swift standard library or Swift Standard Library?
* Swift runtime or Swift Runtime?
* Swift language or Swift Language? The former, I guess, because the name of the language is Swift, not Swift Language.

Ordinary capitalization rules suggest that all except for the last are proper nouns and should be capitalized.

File: .github/ISSUE_TEMPLATE/config.yml (hunk 1 above)
Comment (AnthonyLatsis) [reply .github/ISSUE_TEMPLATE/config.yml:10]: > Maybe "Ask" and "Report" links should be moved up

To the top, before "Discuss an idea" and "Formally propose a change"?

**Comment by AnthonyLatsis** [comment]
I raised the capitalization question with the Contributor Experience Workgroup. We have consensus on "Swift Forums" and "Swift Standard Library", but are not as sure about the rest.

@amartini51 counting on you for a decisive opinion.

**Comment by amartini51** [comment]
> I raised the capitalization question with the Contributor Experience Workgroup. We have consensus on "Swift Forums" and "Swift Standard Library", but are not as sure about the rest.
> 
> @amartini51 counting on you for a decisive opinion.

The documentation style guide calls for "Swift standard library".  The word "Swift" is a proper noun, but the stdlib part of the name is not.  I believe this matches the way many other languages refer to their standard libraries — although my memory might incorrect or partial, since those discussions were circa 2014.  This one has specific guidance from the style guide because misspelling it as "Swift Standard Library" is common.

I don't have a specific style guide entry to point to for "Swift forums" but I would spell it with lowercase f for the same reason.  All of these terms — the Swift compiler, the Swift standard library, the Swift forums, the Swift community — are ordinary nouns (common nouns in the writing sense) being modified by the word "Swift".

**Comment by MaxDesiatov** [comment]
forums.swift.org has this right at the top of the main page:

> The Swift Forums are governed by the Swift Code of Conduct.

Does that mean that this PR should stick to that capitalization, or should we file a bug for Swift forums landing page to be updated?

**Comment by AnthonyLatsis** [comment]
> I don't have a specific style guide entry to point to for "Swift forums" but I would spell it with lowercase f for the same reason. All of these terms — the Swift compiler, the Swift standard library, the Swift forums, the Swift community — are ordinary nouns (common nouns in the writing sense) being modified by the word "Swift".

I definitely make "Swift community" as a common noun. As for the rest, I *kind of* get the point intuitively, but I doubt I could explain this properly to anyone in terms of rules. The logic appears to suggest that "Swift Code of Conduct" is incorrectly capitalized.

> One quick check to help tell common & proper nouns apart is that proper nouns seldom have "a" or "the" in front of them in English — because a proper noun is a name of a specific thing, there's no need for the semantic disambiguation that articles provide. For example, "You wrote a great post in the Swift forums about this feature." is grammatical, but "... in Swift Forums about this..." is odd and stilted. Likewise we write "the Swift compiler" but not "the Swift" or "the Alex".

Personally, I find it more reliable to check whether reformulating the noun as a possessive phrase works just as well, as in "the London Eye" vs. "the eye of London", where it certainly does not. To me it seems like compound nouns that end with a noun that is usually used with an article sound odd without an article in vernacular regardless of whether they are common or proper.

___

Anyway, thank you very much for helping out! I’m really glad to know we have a preferred capitalization for "Swift standard library" that can be extrapolated to similar productions. It would be awesome to eventually publish a subset of the Developer Publications Team’s style guidelines concerned with the Swift project on Swift.org as an official reference for contributors.

**Comment by AnthonyLatsis** [comment]
@swift-ci please smoke test

**Review by xedin — APPROVED** [review]

**Comment by amartini51** [comment]
> Also, the logic appears to suggest that "Swift Code of Conduct" is incorrectly capitalized.

That specific example is a bit of a gray area, so I would have to an editor for help to be completely sure.  That said, "Swift Code of Conduct" is the title of a specific document, so you can write it in title case when referring to that document by name.

**Pull request merged**